internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -286,10 +307,19 @@ func (md *machineDeployment) updateExistingMachines(ctx context.Context, updateE
 	return nil
 }
 
-func (md *machineDeployment) spawnMachineInGroup(ctx context.Context, groupName string, i, total int, standbyFor []string) (string, error) {
+type metadata struct {
+	key   string
+	value string
+}
+
+func (md *machineDeployment) spawnMachineInGroup(ctx context.Context, groupName string, i, total int, standbyFor []string, meta ...metadata) (machine.LeasableMachine, error) {
```

Comment:
Yea, the leasable machine stuff is a bit hard to follow because there's also a goroutine running in the background to constantly renew them. The main driver for the change to this function signature is I need to destroy the canary machine and that function expects a LeasableMachine.

Reply:
I'm having a hard time following the link to the line you're concerned about.

Reply:
This line https://github.com/superfly/flyctl/blob/39195cfab0af1343c1b4724f8ace531656805748/internal/command/deploy/machines_deploymachinesapp.go#L339 in current version of this PR. 

It releases the lease just before returning the leasable machine.

Reply:
What would you suggest? Everything seems to work off of a LeasableMachine object.

Reply:
I mean, the machine is not leased by the time it is returned, that was what I wanted to point out. I guess we can ignore it because the only use case for the returned unleased machine is to be destroyed so it is good anyways.